### PR TITLE
API manager lifecycle Test cases phase 2

### PR DESCRIPTION
**Pull request by sanethd**

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/EditAPIContextAndCheckAccessibilityTestCase.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+import java.util.Map;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Edit the API context and check its accessibility.
+ */
+public class EditAPIContextAndCheckAccessibilityTestCase extends APIManagerLifecycleBaseTest {
+    APIIdentifier apiIdentifierAPI1Version1;
```
Comment (krishanthasamaraweera) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/EditAPIContextAndCheckAccessibilityTestCase.java]: Better to use private access modifier for apiIdentifierAPI1Version1

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/EditAPIContextAndCheckAccessibilityTestCase.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+    private String applicationName;
+    private String newContext;
+    private Map<String, String> requestHeaders;
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+        super.init();
+        apiIdentifierAPI1Version1 = new APIIdentifier(API1_PROVIDER_NAME, API1_NAME, API_VERSION1);
+        applicationName =
```
Comment (krishanthasamaraweera) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/EditAPIContextAndCheckAccessibilityTestCase.java]: Shall we hard code value for api name rather using complex expression to generate api name.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/EditAPIContextAndCheckAccessibilityTestCase.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        assertTrue(oldVersionInvokeResponse.getData().contains(API1_RESPONSE_DATA),
+                "Response data mismatched when invoke  API  before change the context" +
+                        " Response Data:" + oldVersionInvokeResponse.getData());
+
+    }
+
+
+    @Test(groups = {"wso2.am"}, description = "Test changing of the API context",
+            dependsOnMethods = "testInvokeAPIBeforeChangeAPIContext", enabled=false)
+    public void testEditAPIContext() throws Exception {
```
Comment (krishanthasamaraweera) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/EditAPIContextAndCheckAccessibilityTestCase.java]: Throwing generic exception - Can't we handle this.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/EditAPIContextAndCheckAccessibilityTestCase.java (hunk 1 above)
Comment (sanethd) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/EditAPIContextAndCheckAccessibilityTestCase.java]: added private access modifier and variable rename the variable.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/EditAPIContextAndCheckAccessibilityTestCase.java (hunk 3 above)
Comment (sanethd) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/EditAPIContextAndCheckAccessibilityTestCase.java]: done, updated with specific execption.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/EditAPIContextAndCheckAccessibilityTestCase.java (hunk 2 above)
Comment (sanethd) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/EditAPIContextAndCheckAccessibilityTestCase.java]: assuming  this is referring to applicationName, and its now hard coded.

**Pull request merged**